widget_driver/test/drivable_widget_test.dart

```diff
@@ -14,57 +15,149 @@ void main() {
+        final testContainerDrivableWidget = TestContainerDrivableWidget(
+          environmentInfo: _mockRuntimeEnvironmentInfo,
+        );
+        await tester.pumpWidget(MaterialApp(home: testContainerDrivableWidget));
 
-      final driver = testContainerDrivableWidget.driver;
-      expect(driver.runtimeType, TestContainerTestDriver);
+        final driver = testContainerDrivableWidget.driver;
+        expect(driver.runtimeType, TestContainerTestDriver);
+      });
     });
 
-    testWidgets('Can access its driver', (WidgetTester tester) async {
-      when(() => _mockRuntimeEnvironmentInfo.isRunningInTestEnvironment()).thenReturn(false);
+    group('Driver Access: ', () {
+      testWidgets('Can access its driver', (WidgetTester tester) async {
+        when(() => _mockRuntimeEnvironmentInfo.isRunningInTestEnvironment()).thenReturn(false);
 
-      final testContainerDrivableWidget = TestContainerDrivableWidget(
-        environmentInfo: _mockRuntimeEnvironmentInfo,
-      );
-      await tester.pumpWidget(MaterialApp(home: testContainerDrivableWidget));
+        final testContainerDrivableWidget = TestContainerDrivableWidget(
+          environmentInfo: _mockRuntimeEnvironmentInfo,
+        );
+        await tester.pumpWidget(MaterialApp(home: testContainerDrivableWidget));
 
-      final driver = testContainerDrivableWidget.driver;
-      expect(driver.aTestText, 'A test text');
+        final driver = testContainerDrivableWidget.driver;
+        expect(driver.aTestText, 'A test text');
+      });
     });
 
-    testWidgets('Calling `aTestMethod` on driver makes widget reload with new data', (WidgetTester tester) async {
-      when(() => _mockRuntimeEnvironmentInfo.isRunningInTestEnvironment()).thenReturn(false);
+    group('Providable Properties: ', () {
+      testWidgets('Calling `aTestMethod` on driver makes widget reload with new data', (WidgetTester tester) async {
+        when(() => _mockRuntimeEnvironmentInfo.isRunningInTestEnvironment()).thenReturn(false);
+
+        final testContainerDrivableWidget = TestContainerDrivableWidget(
+          environmentInfo: _mockRuntimeEnvironmentInfo,
+        );
+        await tester.pumpWidget(MaterialApp(home: testContainerDrivableWidget));
+        final driver = testContainerDrivableWidget.driver;
+
+        expect(find.text('didCallTestMethod: false'), findsOneWidget);
+
+        driver.aTestMethod();
+        await tester.pump();
 
-      final testContainerDrivableWidget = TestContainerDrivableWidget(
-        environmentInfo: _mockRuntimeEnvironmentInfo,
-      );
-      await tester.pumpWidget(MaterialApp(home: testContainerDrivableWidget));
-      final driver = testContainerDrivableWidget.driver;
+        expect(find.text('didCallTestMethod: true'), findsOneWidget);
+      });
 
-      expect(find.text('didCallTestMethod: false'), findsOneWidget);
+      testWidgets('On state change updateDriverProvidedProperties gets called', (WidgetTester tester) async {
```

Comment:
You mention in a lot of places that on `state changes` we call this `updateDriverProvidedProperties`.
I think this naming is a bit confusing.. What is a "state change"? Is an internal state change of the driver a state change? Is a change of an inherited widget a state change?

When do we actually call it? When the build-method of the `state` of the `statefulwidget` gets called.. 
And when does the state of the stateful widget gets called? Whenever your state calls `setState` or whenever your containing widget gets updated.

Maybe we need to do 2 things here.. 
1: Change the documentation to say: `On widget updates the updateDriverProvidedProperties gets called`
2: Change the code in `drivable_widget.dart` to only call the `widget.driverProvider.updateDriverProvidedProperties(driver);` when the state gets a `didUpdateWidget` call.